tests/unit/test_firewall.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test cases for firewall module."""
+
+from ipaddress import IPv4Network
+
+import pytest
+
+from firewall import Firewall
+
+
+@pytest.mark.parametrize(
+    "domain_ranges, excludes_ranges, expected",
+    [
+        pytest.param(
+            [IPv4Network("10.0.0.0/8")],
```

Comment:
could these be tuples?

Reply:
I made them into lists, since we need to perform `append` operations, converting them to `list` will change the `IPvXNetwork` into `IPvXAddress` and fail the algo unfortunately.